durabletask/internal/grpc_interceptor.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from typing import Any, Dict, Optional
+
+import grpc
+
+
+class UnaryUnaryClientInterceptorImpl (grpc.UnaryUnaryClientInterceptor):
+    """The class implements a UnaryUnaryClientInterceptor from grpc to add an interceptor to add
+    additional headers to all calls as needed."""
+
+    def __init__(self, metadata: Optional[Dict[str, Any]]):
+        super().__init__()
+        self._metadata = metadata
+
+    def intercept_unary_unary(self, continuation, client_call_details, request):
+        if self._metadata:
+            client_call_details = client_call_details._replace(metadata=self._metadata)
```

Comment:
Why are we using a private method here? And what does it do?

Reply:
That was for updating metadata in client_call_details.
I have updated that now.